### PR TITLE
update log types

### LINKED ISSUE #12: refactor: update all debugging/log statements to use tracing
To remain consistent, change all uses of `println!`, `eprintln!`to use the tracing library.

### PR DESCRIPTION
**Pull request by roshaans**
update log types to use tracing

### CODE REVIEW

File: src/metrics.rs [hunk 1]
```diff
@@ -38,13 +38,17 @@ async fn get_metrics() -> impl Responder {
 
     let mut buffer = Vec::new();
     if let Err(e) = encoder.encode(&prometheus::gather(), &mut buffer) {
-        eprintln!("could not encode metrics: {}", e);
+        tracing::error!(target: LOGGING_PREFIX, "could not encode metrics: {}", e);
```
Comment (telezhnaya) [inline src/metrics.rs:41]: Do we now have no appearance of `println/eprintln` in the project?

Comment (roshaans) [reply src/metrics.rs:41]: Yes, this seems the only place we use `println` and `eprintln `statements. Do you see them somewhere else?

Comment (telezhnaya) [reply src/metrics.rs:41]: No, I just wanted to be sure you checked this

**Review by telezhnaya — APPROVED** [review]
Thanks!

I'll merge it later because each commit to `main` triggers cloud build the restart the services. We don't need the restart now

**Pull request merged**